nametoaddr.c

```diff
@@ -126,7 +126,14 @@ pcap_nametoaddr(const char *name)
 	bpf_u_int32 **p;
 	struct hostent *hp;
 
+#ifdef _WIN32
+#pragma warning (push)
+#pragma warning (disable: 4996) /* disable MSVC's gethostbyname() warning here */
+	if ((hp = gethostbyname(name)) != NULL) {
+#pragma warning (pop)
```

Comment:
> If we put the disabling code in a header, the whole following code will disable the warning, which is not what
> we need.

Ok, but how about:
```
+#if defined(BUILDING_LIBPCAP)
+  #define  _WINSOCK_DEPRECATED_NO_WARNINGS
+#endif
+
#include <winsock2.h>
```
in `pcap-stdinc.h`.

> And why we need an #ifdef _MSC_VER? 

Since the pragma is effective for MSVC only. And since the warning is *not* really a warning from the compiler. But a warning caused by *recent* changes in the MS-SDK. Have you tried building with e.g. MinGW and checked the warnings from these unknown pragmas?